launchmon/src/linux/sdbg_linux_std.hxx

```diff
@@ -300,6 +301,51 @@ bool glic_backtrace_wrapper (std::string &bt)
   const T_VA T_UNINIT_HEX                     = 0xdeadbeef;
 # endif // BIT64
 
+#define SDBG_LINUX_DFLT_INSTANTIATION T_VA, \
+                                      T_WT, \
+                                      T_IT, \
+                                      T_GRS,\
+                                      T_FRS,\
+                                      my_thrinfo_t,\
+                                      elf_wrapper
+
+#elif AARCH64_ARCHITECTURE
+
+  // 
+  // 
+  // insert linux AARCH64 architecture macros here ...
+  //
+  //
+  //
+
+# if BIT64
+  //
+  // if the target is 64 bit, use the following
+  //
+  typedef u_int64_t                             T_VA;
+  typedef u_int64_t                             T_WT;
+  typedef u_int32_t                             T_IT;
+  typedef struct user_regs_struct               T_GRS;
+  typedef struct user_fpsimd_struct             T_FRS;
+  const T_IT T_TRAP_INSTRUCTION               = 0x00000000d4200000;
+  const T_IT T_BLEND_MASK                     = 0xffffffff00000000;
+  const T_IT IT_UNINIT_HEX                    = 0xdeadbeefdeadbeefULL;
+  const T_VA T_UNINIT_HEX                     = 0xdeadbeefdeadbeef;
+# else
```

Comment:
Do you know if 32  bit process is supported for ARM? If these are not tested, perhaps it is better to error out?

Reply:
These are untested. I pulled these values from the arm ABI. I agree that generating an error is better.

Reply:
Is static_assert from C++11 allowable here? Or should I define a new macro to create a compile time assert? If I can't use static_assert, where is a good location for utility code?

Reply:
I've been using `# error` preprocessors for this purpose. I think the following will do?

```c
#if AARCH64_ARCHITECTURE
/* Blah blah */
#else
    #error 32-bit target is not supported for the ARM64 architecture
#endif
```

Reply:
Thanks.